src/tagmanager/tm_ctags.c

```diff
@@ -361,22 +383,14 @@ static void rename_anon_tags(TMSourceFile *source_file)
 				if (nested_scope_len <= scope_len)
 					break;
 
-				pos = strstr(nested_tag->scope, orig_name);
 				/* We found the parent name in the nested tag scope - replace it
 				 * with the new name. Note: anonymous tag names generated by
 				 * ctags are unique enough that we don't have to check for
 				 * scope separators here. */
-				if (pos)
-				{
-					gchar *str = g_malloc(nested_scope_len + 50);
-					guint prefix_len = pos - nested_tag->scope;
-
-					strncpy(str, nested_tag->scope, prefix_len);
-					strcpy(str + prefix_len, new_name);
-					strcpy(str + prefix_len + new_name_len, pos + orig_name_len);
-					g_free(nested_tag->scope);
-					nested_tag->scope = str;
-				}
+				replace_str(&nested_tag->scope, orig_name, orig_name_len, new_name, new_name_len);
+
+				/* Do the same for var_type as well */
+				replace_str(&nested_tag->var_type, orig_name, orig_name_len, new_name, new_name_len);
```

Comment:
Shouldn't this new helper be used around L406?

Reply:
Right, I missed we are doing the same replacement once more below. We could maybe even use the GString version, I'm not sure if the performance is too critical here.

I checked what you propose and it looks good to me and unifies the code a little.